File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    let answer = [];
    let honor = [];
    for(let i = 0 ; i < score.length ; i++){
        if(honor.length < k) honor.push(score[i]);
        else {
            let min = Math.min(...honor)
            if(score[i] > min){
                honor.pop(min);
                honor.push(+score[i]);
            }
        }
        honor.sort((a, b) => b-a);
        answer.push(+(honor.slice(-1)))
    }
    return answer;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    honor = [];
    for i in range(len(score)):
        if len(honor) < k:
            honor.append(score[i])
        else:
            if min(honor) < score[i]:
                del honor[-1]
                honor.append(score[i])
                
        honor.sort(reverse=True)
        answer.append(honor[-1])
    return answer